### PR TITLE
DataLabelWidget: Format numbers language aware

### PR DESCRIPTION
**Pull request by apepper**
Formats a numbers attribute according to the language of a site. To test you could use the `size` attribute of a Document.

### CODE REVIEW

File: src/Widgets/DataLabelWidget/DataLabelWidgetComponent.tsx [hunk 1]
```diff
@@ -82,6 +86,10 @@ function Text({ value }: { value: unknown }) {
   return value ? value.toString() : localizeNotAvailable()
 }
 
+function NumberText({ value }: { value: number }) {
+  return new Intl.NumberFormat(getCurrentLanguage()).format(value)
```
Comment (dcsaszar) [inline src/Widgets/DataLabelWidget/DataLabelWidgetComponent.tsx:90]: * [ ] FYI From experience users often prefer `{useGrouping: "min2"}` to keep four-digit numbers as a whole for better readability.

**Review by dcsaszar — APPROVED** [review]
Good, but see comment.

File: src/Widgets/DataLabelWidget/DataLabelWidgetComponent.tsx (hunk 1 above)
Comment (apepper) [reply src/Widgets/DataLabelWidget/DataLabelWidgetComponent.tsx:90]: This seems to be a fairly recent addition to the standard. At least according to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat Safari only supports it with 16.4 or newer (which was released last year).

Comment (dcsaszar) [reply src/Widgets/DataLabelWidget/DataLabelWidgetComponent.tsx:90]: A workaround for older browsers is `useGrouping: Math.abs(value) >= 10000`.

**Comment by apepper** [comment]
REVIEW ([2c53be2](https://github.com/Scrivito/scrivito-portal-app/pull/441/commits/2c53be235281cfe372505821efe679eaa1460e9c))

**Pull request merged**